contracts/cosmwasm-vm/cw-common/src/xcall_connection_msg.rs

```diff
@@ -2,6 +2,7 @@ use crate::cw_types::{
     CwChannelCloseMsg, CwChannelConnectMsg, CwChannelOpenMsg, CwPacketAckMsg, CwPacketReceiveMsg,
```

Comment:
Anyway you can extend in rust? so that xCall connection extends the ibc-app message?

Reply:
Only trait inheritance is allowed in rust. For enums and objects we would need to use composition like 
`enum XcallConnMsg {
IbcMsg(IbcMsgType)
}` but this breaks ibc host since it assumes each ibc message to have separate endpoint. The composition looks better by design but might need refactoring on ibc host as well .

Reply:
Okey lets keep it as for now then